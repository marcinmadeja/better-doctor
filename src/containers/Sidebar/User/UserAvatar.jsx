// @flow strict
import * as React from 'react';
import styled, { css } from 'styled-components';
import { withStyles } from '@material-ui/core';
import Avatar from '@material-ui/core/Avatar';
import { COLORS } from 'config/theme';

const StyledUserAvatar = styled.div`
  position: relative;
  display: inline-block;
`;

type StatusProps = {
  status?: string,
};

const Status = (styled.div`
  width: 11px;
  height: 11px;
  border-radius: 100%;
  position: absolute;
  bottom: 3px;
  right: 3px;

  ${props => props.status === 'active' && css`
    background-color: ${COLORS.success};
  `}

  ${props => props.status === 'inactive' && css`
    background-color: ${COLORS.failure};
  `}
`: React.ComponentType<StatusProps>);

const styles = {
  bigAvatar: {
    width: 50,
    height: 50,
  },
};

type Props = {
  src: string,
  alt?: string,
  status: string,
  classes: { [string]: string },
};

const UserAvatar = ({ src, alt, status, classes }: Props) => {
  return (
    <StyledUserAvatar>
      <Avatar src={src} alt={alt} className={classes.bigAvatar} />

      <Status status={status} />
    </StyledUserAvatar>
  );
};

UserAvatar.defaultProps = {
  alt: '',
};

export default withStyles(styles)(UserAvatar);
