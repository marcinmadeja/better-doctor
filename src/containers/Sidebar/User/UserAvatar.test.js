import React from 'react';
import 'jest-styled-components';
import UserAvatar from './UserAvatar';

const defaultProps = {
  src: 'image/src/',
  alt: 'image alt',
  status: 'active',
  classes: {},
};

const setup = (props = {}) => {
  props = { ...defaultProps, ...props };
  const actions = {};
  const component = <UserAvatar {...actions} {...props} />;
  const shallowComponent = shallow(component);

  return {
    actions,
    component,
    shallowComponent,
  };
};

describe('UserAvatar', () => {
  it('should render', () => {
    const { shallowComponent } = setup();
    expect(shallowComponent.exists()).toBe(true);
  });

  it('should match main snapshot', () => {
    const { component } = setup();
    const tree = renderer
      .create(component)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
