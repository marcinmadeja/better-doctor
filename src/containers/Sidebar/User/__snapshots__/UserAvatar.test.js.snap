// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar should match main snapshot 1`] = `
.c0 {
  position: relative;
  display: inline-block;
}

.c1 {
  width: 11px;
  height: 11px;
  border-radius: 100%;
  position: absolute;
  bottom: 3px;
  right: 3px;
  background-color: #0aa699;
}

<div
  className="c0"
>
  <div
    className="MuiAvatar-root-2 UserAvatar-bigAvatar-1"
  >
    <img
      alt="image alt"
      className="MuiAvatar-img-4"
      sizes={undefined}
      src="image/src/"
      srcSet={undefined}
    />
  </div>
  <div
    className="c1"
  />
</div>
`;
